File: src/com/sterndu/json/JsonArray.java
package com.sterndu.json;

import java.util.*;
import java.util.function.Function;

public class JsonArray extends ArrayList<Object> implements JsonValue {

	/**
	 *
	 */
	private static final long serialVersionUID = 2915477326174591243L;

	public JsonArray() {}

	public JsonArray(Collection<?> c) {
		super(c);
	}

	@Override
	public String toJson() {
		return toJson(o -> "\"" + o + "\"");
	}

	@Override
	public String toJson(Function<Object, String> function) {
		StringBuilder sb = new StringBuilder();
		sb.append("[\n");
		boolean b = true;
		for (Object o: this) if (o!=this) {
			if (b) b = false;
			else sb.append(",\n");
			sb.append(JsonUtil.serialize(o, function));
		}
		return sb.toString().replace("\n", "\n\t") + "\n]";
	}

	@Override
	public JsonValue toJsonValue() {
		return this;
	}

	@Override
	public JsonValue toJsonValue(Function<Object, String> function) {
		return this;
	}

	@Override
	public String toString() {
		return toJson();
	}


}

File: src/com/sterndu/json/JsonArray.kt
@file:JvmName("JsonArray")
package com.sterndu.json

class JsonArray : ArrayList<Any?>, JsonValue {
	constructor() : super()
	constructor(c: Collection<*>) : super(c)

	override fun toJson(): String {
		return toJson { "\"$it\"" }
	}

	override fun toJson(function: (Any?) -> String): String {
		val sb = StringBuilder()
		sb.append("[\n")
		var b = true
		for (o in this) if (o != this) {
			if (b) b = false else sb.append(",\n")
			sb.append(serialize(o, function))
		}
		return sb.toString().replace("\n", "\n\t") + "\n]"
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}

	companion object {
		private const val serialVersionUID = 2915477326174591243L
	}
}
